### PR TITLE
auth

### PR DESCRIPTION
**Pull request by Gavrilik**

### CODE REVIEW

File: src/features/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Controller, Post, UseGuards, Request } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+
+@Controller()
```
Comment (dmialeshka) [inline src/features/auth/auth.controller.ts:4]: add 'auth' path

File: src/features/user/user.module.ts [hunk 2]
```diff
@@ -2,9 +2,10 @@ import { Module } from '@nestjs/common';
 import { UserService } from './user.service';
 import { UserController } from './user.controller';
 import { UserJsonRepository } from './user.json-repository';
+import { AuthController } from '../auth/auth.controller';
 
 @Module({
-  controllers: [UserController],
+  controllers: [UserController, AuthController],
```
Comment (dmialeshka) [inline src/features/user/user.module.ts:8]: Why UserModule respond for AuthController

File: src/features/user/user.service.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import { Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
-import { User } from './entities/user.entity';
 import { UserJsonRepository } from './user.json-repository';
 @Injectable()
 export class UserService {
+  findByEmail(email: string) {
```
Comment (dmialeshka) [inline src/features/user/user.service.ts:7]: move under constructor regarding CRUD operation

**Review by dmialeshka — CHANGES_REQUESTED** [review]

File: src/features/user/user.json-repository.ts [hunk 4]
```diff
@@ -11,6 +9,9 @@ import defaulteUsers from './user.json'; //массив, работа с мас
 let users = defaulteUsers;
 @Injectable()
 export class UserJsonRepository {
+  findByEmail(email: string) {
```
[the remark this answers is not in the document]
Comment (Gavrilik) [reply src/features/user/user.json-repository.ts:12]: done

File: src/features/user/user.module.ts (hunk 2 above)
Comment (Gavrilik) [reply src/features/user/user.module.ts:8]: he is not needed here, deleted

File: src/features/user/user.service.ts (hunk 3 above)
Comment (Gavrilik) [reply src/features/user/user.service.ts:7]: moved

File: src/features/auth/auth.controller.ts (hunk 1 above)
Comment (Gavrilik) [reply src/features/auth/auth.controller.ts:4]: added

Comment (dmialeshka) [reply src/features/auth/auth.controller.ts:5]: Where ?

Comment (Gavrilik) [reply src/features/auth/auth.controller.ts:5]: correct

File: src/features/auth/auth.service.ts [hunk 5]
```diff
@@ -5,10 +5,10 @@ import { UserService } from '../user/user.service';
 export class AuthService {
   constructor(private userService: UserService) {}
 
-  async validateEmail(email: string, pass: string): Promise<any> {
+  async validateUser(email: string, pass: string): Promise<any> {
     const user = await this.userService.findByEmail(email);
     if (user && user.password === pass) {
```
Comment (dmialeshka) [inline src/features/auth/auth.service.ts:10]: handle bad case first

File: src/features/user/user.controller.ts [hunk 6]
```diff
@@ -6,12 +6,15 @@ import {
...
   Delete,
+  UseGuards,
 } from '@nestjs/common';
 import { UserService } from './user.service';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
+import { AuthGuard } from '@nestjs/passport';
 
 @Controller('user')
+@UseGuards(AuthGuard('local'))
```
Comment (dmialeshka) [inline src/features/user/user.controller.ts:17]: Set guard for create, update and delete operations, instead all

**Review by dmialeshka — CHANGES_REQUESTED** [review]

File: src/features/auth/auth.module.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { Module } from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { PassportModule } from '@nestjs/passport';
+import { LocalStrategy } from './local.strategy';
+import { UserModule } from '../user/user.module';
+import { UserController } from '../user/user.controller';
+import { AuthController } from './auth.controller';
+
+@Module({
+  controllers: [AuthController, UserController],
```
Comment (dmialeshka) [inline src/features/auth/auth.module.ts:10]: Why AuthModule respond for UserController

File: src/features/user/user.controller.ts (hunk 6 above)
Comment (Gavrilik) [reply src/features/user/user.controller.ts:17]: done

File: src/features/auth/auth.module.ts (hunk 7 above)
Comment (Gavrilik) [reply src/features/auth/auth.module.ts:10]: correct

**Review by dmialeshka — APPROVED** [review]

**Pull request merged**